### ISSUE TITLE
pgbackrest 2.23 version connection timeout and retry "RequestTimeTooSkewed" issues

### ISSUE THREAD

**Issue by krishna-kurapati** [pgbackrest/pgbackrest#1178]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: pgBackRest 2.23

2. PostgreSQL version: 11.6.1
 PostgreSQL 11.6 (VMware Postgres 11.6.1) on x86_64-redhat-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-39)
, 64-bit

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:

4. Did you install pgBackRest from source or from a package? from a package
[pgbackrest.log](https://github.com/pgbackrest/pgbackrest/files/5248277/pgbackrest.log)
5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)  -- attached
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`) -- attached
[postgresql.log](https://github.com/pgbackrest/pgbackrest/files/5248278/postgresql.log)

    - errors in the postgresql log file before or during the time you experienced the issue -- no errors
[server_log_Sun.log](https://github.com/pgbackrest/pgbackrest/files/5248453/server_log_Sun.log)
[server_log_Sun01.log](https://github.com/pgbackrest/pgbackrest/files/5248454/server_log_Sun01.log)
[server_log_01Sat.log](https://github.com/pgbackrest/pgbackrest/files/5248455/server_log_01Sat.log)
[server_log_Sat.log](https://github.com/pgbackrest/pgbackrest/files/5248456/server_log_Sat.log)

    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`) -- attached
[pgbackrest_log_12_13sep.zip](https://github.com/pgbackrest/pgbackrest/files/5248472/pgbackrest_log_12_13sep.zip)

7. Describe the issue:
Team, I have attached all the required files. We are facing connection timed out and RequestTimeTooSkewed errors and for each error occurrence the sessions are going away for 16 minutes and again resuming the backup process which causes the backup to complete 18 hours of 1000GB sized database. Please see the below error for your quick action. Please check and help me to fix this issue. and the same errors are occurring on every full/diff backup runs.

2020-09-12 22:00:44.570 P00  DEBUG:     storage/storage::storageNewWrite: => {type: s3, name: {"/data1/postgresdb-backups/backup/prod/20200912-215901F/backup.manifest.copy"}, modeFile: 0000, modePath: 0000, createPath: true, syncFile: true, syncPath: true, atomic: true}
2020-09-12 22:00:44.570 P00  DEBUG:     info/manifest::manifestSave: (this: {Manifest}, write: {IoWrite})
2020-09-12 22:00:44.572 P00  DEBUG:     info/info::infoSave: (this: {Info}, write: {IoWrite}, callbackFunction: (function *), callbackData: *void)
**2020-09-12 22:16:13.652 P00  DEBUG:     common/io/http/client::httpClientRequest: retry KernelError: tls failed syscall: [110] Connection timed out
2020-09-12 22:16:13.688 P00  DEBUG:     storage/s3/storage::storageS3Request: retry RequestTimeTooSkewed: The difference between the request time and the current time is too large.**
2020-09-12 22:16:13.829 P00  DEBUG:     info/info::infoSave: => void
2020-09-12 22:16:13.829 P00  DEBUG:     info/manifest::manifestSave: => void
2020-09-12 22:16:13.829 P00  DEBUG:     command/backup/backup::backupManifestSaveCopy: => void
2020-09-12 22:16:13.829 P00  DEBUG:     command/backup/backup::backupJobResult: (manifest: {Manifest}, host: null, fileName: {"/data1/data_prod/base/20686/94771"}, fileRemove: {[]}, job: {state: done, key: {"pg_data/base/20686/94771"}, command: {command: backupFile}, code: 0, message: null, result: {VariantList}}, sizeTotal: 1074038461627, sizeCopied: 11811160064, pageSize: 8192)
2020-09-12 22:16:13.829 P05   INFO: backup file /data1/data_prod/base/20686/94771 (1GB, 1%) checksum f2aa40357bc7c2acae4c70ee1333156bcd52c44d

Thanks
Krishna

**Comment by dwsteele** [comment]
As far as we can tell this was caused by a kernel update -- see https://github.com/pgbackrest/pgbackrest/issues/940 for details (warning, it's a long thread).

Long story short, we switched to a non-blocking TLS implementation for 2.26 that works on older and newer kernels. So, the answer is to upgrade to 2.26.

Since it looks like you are getting a bundle from VMware you'll need to contact them directly.
